### PR TITLE
refactor: api-v2 create basic players resource

### PR DESCRIPTION
**Pull request by alowave223**
Referring to the https://github.com/akatsuki-v2/users-service/tree/main I made basic player and response models for v2, also I'm not sure where i put the models and responses folder.

### CODE REVIEW

File: app/api/v2/models/players.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+    custom_badge_name: Optional[str]
+    custom_badge_icon: Optional[str]
+
+    userpage_content: Optional[str]
+
+
+class Players(BaseModel):
+    players: List[Player]
+    max_pages: int
+    current_page: int
```
Comment (cmyui) [inline app/api/v2/models/players.py:37]: this abstraction should be at the level of a successful response, that way you don't need this on a per resource basis

will show you what i mean in a commit soon

File: app/api/v2/players.py [hunk 2]
```diff
@@ -1,11 +1,52 @@
...
-    return f"Hello! {player_id}"
+@router.get("/players", response_model=Success[Players])
+async def get_players(
+    page: Optional[int] = Query(1, ge=1),
+    page_size: Optional[int] = Query(50, ge=50, le=100),
+):
+    players = await players_repo.fetch_many(page=page, page_size=page_size)
+
+    if len(players) < 1:
+        return failure("Reached the pages limit.")
```
Comment (cmyui) [inline app/api/v2/players.py:29]: no need for this, just return an empty list

File: app/api/v2/players.py [hunk 3]
```diff
@@ -1,11 +1,52 @@
...
+    )
+    return success(response)
+
+
+@router.get("/players/{player}", response_model=Success[Player])
+async def get_player(player: Union[int, str]):
+    if isinstance(player, int):
+        player_info = await players_repo.fetch_one(id=player)
+    else:
+        player_info = await players_repo.fetch_one(name=player)
```
Comment (cmyui) [inline app/api/v2/players.py:46]: we won't have support for lookups by name, only id

File: app/api/v2/players.py [hunk 4]
```diff
@@ -1,11 +1,52 @@
...
+
+    if player_info is None:
+        return failure("Player not found.", 404)
```
Comment (cmyui) [inline app/api/v2/players.py:49]: for http status codes, the enum in `fastapi.status` should be preferred

**Comment by cmyui** [comment]
there isn't much value in splitting responses up, as we will not be adding additional responses - there will only be success, and error responses

File: app/api/v2/players.py [hunk 5]
```diff
@@ -1,11 +1,52 @@
...
+
+    response = Players(
+        players=[Player.from_mapping(x) for x in players],
+        max_pages=players_count / page_size + 1,
+        current_page=page,
+    )
+    return success(response)
+
+
+@router.get("/players/{player}", response_model=Success[Player])
```
Comment (cmyui) [inline app/api/v2/players.py:41]: this is *very* new, but as of fastapi 0.89.0, `response_model` can be replaced with a return type annotation
https://github.com/tiangolo/fastapi/releases/tag/0.89.0

File: app/api/v2/models/players.py [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+from typing import Optional
+
+from . import BaseModel
+
+
+class Player(BaseModel):
+    id: int
+    name: str
+    safe_name: str
+    _email: str
```
Comment (cmyui) [inline app/api/v2/models/players.py:13]: any reason for the underscore?

File: app/api/v2/players.py [hunk 7]
```diff
@@ -1,11 +1,52 @@
...
 router = APIRouter()
 
 
-@router.get("/players/{player_id}")
-async def get_player(player_id: int):
-    return f"Hello! {player_id}"
+@router.get("/players", response_model=Success[Players])
+async def get_players(
+    page: Optional[int] = Query(1, ge=1),
+    page_size: Optional[int] = Query(50, ge=50, le=100),
```
Comment (cmyui) [inline app/api/v2/players.py:24]:
```suggestion
    page_size: Optional[int] = Query(50, ge=1, le=100),
```

**Comment by cmyui** [comment]
looks pretty good overall! nice work so far

File: app/api/v2/models/players.py (hunk 6 above)
Comment (alowave223) [reply app/api/v2/models/players.py:13]: Exclude email to be publicly available

**Comment by cmyui** [comment]
going to take over pr for a bit

**Comment by cmyui** [comment]
let's merge this for now and figure out PATCH/DELETE later once we have auth in place

**Review by cmyui — APPROVED** [review]

**Pull request merged**